server/src/repositories/AuditLogs/AuditLogRepository.ts

```diff
@@ -0,0 +1,74 @@
+import {AuditLogs} from "../../models/auditLogs/auditLogs";
+import { knex } from "../../db";
+import {singleLogToDomain, logsToDomain} from "../../mappers/auditLogs/auditLogMapper"
+
+export interface IAuditLogRepo {
+    getLogByID(logID: number): Promise<AuditLogs | null>;
+    getLogs(): Promise<AuditLogs[]>;
+    addLog(log: AuditLogs): Promise<AuditLogs | null>;
+    modifyLogDescription(logID: number, description: string): Promise<AuditLogs | null>;
+    deleteLog(logID: number): Promise<void>;
+}
+export  class  AuditLogRepo implements IAuditLogRepo {
+
+    private queryBuilder;
+
+    constructor(queryBuilder?: any) {
+        this.queryBuilder = queryBuilder || knex;
+    }
+
+    public async getLogByID(logID: number): Promise<AuditLogs | null> {
+        const knexResult = await this.queryBuilder
+            .first(
+                "id",
+                "timeDate",
+                "user",
+                "eventType",
+                "description"
+            )
+            .from("AuditLogs")
+            .where("id", logID);
+
+        return singleLogToDomain(knexResult);
+    }
+
+    public async getLogs(): Promise<AuditLogs[]> {
+        const knexResult = await this.queryBuilder("InventoryItem").select(
+            "AuditLogs.id",
+            "AuditLogs.timeDate",
```

Comment:
You mean like 36, that is 100% overlook on my part with "InventoryItem" instead of "AuditLogs"
fixing that now and going to overlook everything else one more time.